Lib/test/test_builtin.py

```diff
@@ -218,6 +218,16 @@ def test_all(self):
         S = [50, 40, 60]
         self.assertEqual(all(x > 42 for x in S), False)
 
+    def test_all_equal(self):
+        self.assertEqual(all_equal([1, 1, 1]), True)
+        self.assertEqual(all_equal([1, 2, 3]), False)
+        self.assertEqual(all_equal([1, None, 1]), False)
+        self.assertRaises(RuntimeError, all_equal, TestFailingIter())
+        self.assertRaises(TypeError, all_equal, 10)                 # Non-iterable
+        self.assertRaises(TypeError, all_equal)                     # No args
+        self.assertRaises(TypeError, all_equal, [2, 4, 6], [])      # Too many args
+        self.assertEqual(all_equal([]), True)                       # Empty iterator
```

Comment:
```suggestion
        self.assertTrue(all_equal([1, 1, 1]))
        self.assertFalse(all_equal([1, 2, 3]))
        self.assertFalse(all_equal([1, None, 1]))
        self.assertRaises(RuntimeError, all_equal, TestFailingIter())
        self.assertRaises(TypeError, all_equal, 10)                 # Non-iterable
        self.assertRaises(TypeError, all_equal)                     # No args
        self.assertRaises(TypeError, all_equal, [2, 4, 6], [])      # Too many args
        self.assertTrue(all_equal([]))                              # Empty iterator
```

Reply:
Thanks, good suggestion. I was mostly following the existing code in the file here.

Reply:
Should be resolved now